### DISCUSSION TITLE
Tuya power plug metering issue

### DISCUSSION THREAD

**Discussion by Gerwintje** [Koenkk/zigbee2mqtt#16982] [Devices]
Hi all, 

Recently bought a couple of Tuya smart plugs (https://www.amazon.nl/-/en/dp/B09SG3VYW6?ref=ppx_yo2ov_dt_b_product_details&th=1). Adding them to Z2M went flaweless. Adding them to HA is without issue. However, I think I'm missing certain sensors. 

For example the power metering sensor (Power) is in HA under 'Diagonose'. The value of Power is updated as expected. This is also shown in Z2M log. 

The Plug is seen under Z2M as follows;

        "manufacturerName": "_TZ3000_gjnozsaz",
        "model": "TS011F_plug_1",

Tried updating firmware, tried re-adding/interviewing, but no succes. it _seems_ that there is no issue with Z2M, but cannot confirm. Only ting I CAN confirm is that values are being updated in Z2M (like power) but HA is missing/mis interpreting these values.

Setup is as follow; Z2M in docker → Mosquitto Broker in docker → HA in docker. 

Other Zigbee devices are not giving issues, only these plugs (have 4 of them, alle with same result). 

Any ideas?

**Comment by lux73** [comment]
there is NO issue with z2m:

About:
```
Friendly name
    tuya-waschkeller
Description

Last seen
    just now
Availability
    [Disabled](https://www.zigbee2mqtt.io/guide/configuration/device-availability.html#availability-advanced-configuration)
Device type
    Router
Zigbee Model
    TS011F
Zigbee Manufacturer
    _TZ3000_gjnozsaz
Description
    Smart plug (with power monitoring)
Support status

    Supported
IEEE Address
    0xa4c1383ec0xxxxxx
Network address
    0x65AD
Manufacturer
    [TuYa](https://www.zigbee2mqtt.io/supported-devices/#v=TuYa)
Model
    [TS011F_plug_1](https://www.zigbee2mqtt.io/devices/TS011F_plug_1.html#tuya-ts011f_plug_1)
Power
Interview completed
    True
```

State:
```
{
    "child_lock": "UNLOCK",
    "current": 7.38,
    "energy": 3.39,
    "indicator_mode": "off",
    "last_seen": 1678520055929,
    "linkquality": 0,
    "power": 1784,
    "state": "ON",
    "update": {
        "installed_version": 192,
        "latest_version": 192,
        "state": "idle"
    },
    "voltage": 224,
    "power_outage_memory": "on"
}
```

**Comment by kzajac83** [comment]
If good unserstood @Gerwintje then have the same problem but after investigation seems issue is in reporting/history mapping.

1st of all earlier for socket "TS011F_plug_1" I was usisng ZHA in HA and expect #16076 general no issues.   Today switch integration ZHA->Z2M and in exposes in Z2M tab I saw correct values like power, current, voltage:

![image](https://user-images.githubusercontent.com/50861081/231574952-aee7c083-a9b4-4cd4-92d0-9685f0272613.png)

In ZHA integration all mention values was assing to the device so simple choose my socket in history i was able see all what I need but in Z2M missing mention values in history under the device. So when I go to history tab of HA and choose as device my socekt then see this (please igone gaps, restart HA/Z2M many times in meantime):

![image](https://user-images.githubusercontent.com/50861081/231575322-4e143d3c-ac18-4872-a3db-255bed4f216c.png)

Missing voltage, currect, power.  What is intersted this values exist in entities but not connect to device - this is main my issue:

![image](https://user-images.githubusercontent.com/50861081/231575897-d988606f-0003-4137-bb09-6335f6eb8b59.png)

And I can draw history using above entities:
![image](https://user-images.githubusercontent.com/50861081/231576384-c942c78f-b016-46b7-9b25-e5bb80062a66.png)

![image](https://user-images.githubusercontent.com/50861081/231576287-2a72b59f-d509-41ea-a6ff-5dddb84d4efe.png)


Andthe same what mention creator of this issue, why voltage, current, power are in diagnostic section? Should be in sensos. Moving this data to sensors should be resolve the issue

![image](https://user-images.githubusercontent.com/50861081/231576681-14ec029b-6008-402c-997e-2ec276569fae.png)

 Also please note diagnostic values entitiesby defaultis turn off, I was need manually turn it on.  But what is strage history is independent of state of this entitiies!  I was able draving history for turn off volateges, current. 

As mention on begging  general something wrong is classification entities in diagnostic instead of in sensors.

**Reply by ArnoldDeRuiter** [reply]
Thank you! I hope this can be rectified at some point, so the energy measurement section can work with this as well.
